client/src/views/Authentication/AuthEmailAction.vue

```
@@ -0,0 +1,118 @@
+<template>
+  <div class="q-pa-md">
+    <div class="q-gutter-y-sm column fixed register" style="width: 300px" v-if="reset">
+      <p style="text-align: center; margin-top:20px; margin-bottom: 0">Create New Password</p>
+
+        <q-input v-model.lazy="password" :type="isPwd1 ? 'password' : 'text'" label="Password" :dense="dense" :rules="[val => checkPasswordField(val)]">
+            <template v-slot:append>
+            <q-icon
+                :name="isPwd1 ? 'visibility_off' : 'visibility'"
+                class="cursor-pointer"
+                @click="isPwd1 = !isPwd1"
+            />
+            </template>
+        </q-input>
+
+        <q-input v-model.lazy="passwordConfirm" :type="isPwd2 ? 'password' : 'text'" label="Confirm Password" :dense="dense" :rules="[val => checkRepeatPasswordField(val)]">
+            <template v-slot:append>
+            <q-icon
+                :name="isPwd2 ? 'visibility_off' : 'visibility'"
+                class="cursor-pointer"
+                 @click="isPwd2 = !isPwd2"
+            />
+            </template>
+        </q-input>
+
+        <q-btn color="white" text-color="black" @click="resetPass" :disabled="!enableCreate">Set password
+            <q-spinner-bars
+            color="primary"
+            size="1em"
+            v-show="loading"
+            />
+        </q-btn>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from 'axios'
+import { validatePassword } from '../../utilities/auth.js'
+
+export default {
+  name: 'register',
+  data () {
+    return {
+      password: '',
+      passwordConfirm: '',
+      dense: false,
+      isPwd1: true,
+      isPwd2: true,
+      loading: false,
+      reset: false,
+      passwordField: false,
+      oobCode: ''
+    }
+  },
+
+  methods: {
+    checkPasswordField: function (val) {
+      if (validatePassword(val) && validatePassword(val).message === 'empty password') {
+        this.passwordField = false
+        return 'Password field cannot be empty'
```

You can simplify logic here, instead of comparing the message value to a string you can simply return that error message from _validatePassword_ function